### PR TITLE
FT2023-339: Implementation of the first custom module.

### PR DESCRIPTION
**Pull request by Arijit-Acharya-innoraft**
… files, also their controller and routing files.

Implemented hook_cron and hook_form_alter in the  .module file.

### CODE REVIEW

File: web/modules/custom/mymodule/mymodule.info.yml [hunk 1]
```diff
@@ -0,0 +1,8 @@
+name: 'My Module'
+description: 'This is the first custom module.'
+type: module
+core_version_requirement: ^10
+dependencies:
```
Comment (vivekpan27) [inline web/modules/custom/mymodule/mymodule.info.yml]: Are these dependencies required?

File: web/modules/custom/mymodule/mymodule.module [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+*Implements hook_cron
+*/
+function mymodule_cron() {
+  \Drupal::messenger()->addMessage(t("Hello World from mymodule  hook_cron"));
+}
+
+/**
+* This implements hook_form_alter().
+*/
+function mymodule_form_alter(&$form, \Drupal\Core\Form\FormStateInterface $form_state, $form_id) {
```
Comment (vivekpan27) [inline web/modules/custom/mymodule/mymodule.module]: Please check in Drupal core how the form parameters for form_alter are mentioned.

File: web/modules/custom/mymodule/mymodule.module [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+  //  \Drupal::messenger()->addMessage(t("the form id of this form is @id:",['@id'=>$form_id]));
```
Comment (vivekpan27) [inline web/modules/custom/mymodule/mymodule.module]: Please remove comment.s

File: web/modules/custom/mymodule/src/Controller/FirstController.php [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+use Drupal\Core\Controller\ControllerBase;
+use Drupal\Core\Cache;
+/**
+ * This class is a multipurpose utility class.
+ * it is used for testing simple markups,getAccountName function.  
+ */
+class FirstController extends ControllerBase {
+
+  /**
+   * @return [array]
```
Comment (vivekpan27) [inline web/modules/custom/mymodule/src/Controller/FirstController.php]: Please check in Drupal core how the function docblock is created.
This way is incorrect.

File: web/modules/custom/mymodule/src/Form/FirstForm.php [hunk 5]
```diff
@@ -0,0 +1,85 @@
+<?php
+
+namespace Drupal\mymodule\Form;
+
+use Drupal\Core\Form\FormBase;
+use Drupal\Core\Form\FormStateInterface;
+
+class FirstForm extends FormBase {
+  
+  function getFormId() {
```
Comment (vivekpan27) [inline web/modules/custom/mymodule/src/Form/FirstForm.php]: Where is the function docblock for this function and others?

File: web/modules/custom/mymodule/src/Form/FirstForm.php [hunk 6]
```diff
@@ -0,0 +1,85 @@
+<?php
+
+namespace Drupal\mymodule\Form;
+
+use Drupal\Core\Form\FormBase;
+use Drupal\Core\Form\FormStateInterface;
+
+class FirstForm extends FormBase {
```
Comment (vivekpan27) [inline web/modules/custom/mymodule/src/Form/FirstForm.php:11]: Where is the docblock for the class?

File: web/modules/custom/mymodule/src/Form/FirstForm.php [hunk 7]
```diff
@@ -0,0 +1,85 @@
...
+    $form['submit'] = [
+      '#type' => 'submit',
+      '#value' => $this->t('Submit'),
+    ];
+
+    return $form;
+  }
+
+  function validateForm(array &$form, FormStateInterface $form_state) {
+    $publicDomains = [];
```
Comment (vivekpan27) [inline web/modules/custom/mymodule/src/Form/FirstForm.php]: All local variables should be camelcase.
Only class variables should be SnakeCased.

File: web/modules/custom/mymodule/src/Form/FirstForm.php (hunk 6 above)
Comment (vivekpan27) [inline web/modules/custom/mymodule/src/Form/FirstForm.php:11]: The question in the ticket mentions to create a config form.
Is this a config form?

File: web/modules/custom/mymodule/mymodule.info.yml (hunk 1 above)
Comment (Arijit-Acharya-innoraft) [reply web/modules/custom/mymodule/mymodule.info.yml]: For the sake of exploring , I have added them.

File: web/modules/custom/mymodule/mymodule.module (hunk 3 above)
Comment (Arijit-Acharya-innoraft) [reply web/modules/custom/mymodule/mymodule.module]: Done.

File: web/modules/custom/mymodule/mymodule.module (hunk 2 above)
Comment (Arijit-Acharya-innoraft) [reply web/modules/custom/mymodule/mymodule.module]: Done

File: web/modules/custom/mymodule/src/Form/FirstForm.php (hunk 6 above)
Comment (Arijit-Acharya-innoraft) [reply web/modules/custom/mymodule/src/Form/FirstForm.php:11]: Sir in ticket FT2023-341 we are asked to do the config form. This ticket was to explore custom form, so just made a simple form.

Comment (Arijit-Acharya-innoraft) [reply web/modules/custom/mymodule/src/Form/FirstForm.php:11]: Added

File: web/modules/custom/mymodule/src/Controller/FirstController.php (hunk 4 above)
Comment (Arijit-Acharya-innoraft) [reply web/modules/custom/mymodule/src/Controller/FirstController.php]: Rectified.

File: web/modules/custom/mymodule/src/Form/FirstForm.php (hunk 5 above)
Comment (Arijit-Acharya-innoraft) [reply web/modules/custom/mymodule/src/Form/FirstForm.php]: Added

**Review by Arijit-Acharya-innoraft — COMMENTED** [review]
@vivekpan27  Made the required changes. Please Review it.

File: web/modules/custom/mymodule/mymodule.info.yml (hunk 1 above)
Comment (vivekpan27) [reply web/modules/custom/mymodule/mymodule.info.yml]: We refrain from including unnecessary code in the repository.

File: web/modules/custom/mymodule/mymodule.module [hunk 8]
```diff
@@ -0,0 +1,30 @@
...
+/**
+ * @file
+ */
+
+use Drupal\Core\Form\FormStateInterface;
+
+/**
+ * @file
+ * This file implements how to implement the hoook function.
+ */
```
Comment (vivekpan27) [inline web/modules/custom/mymodule/mymodule.module]: Why do I see 2 @file statements?

File: web/modules/custom/mymodule/src/Controller/FirstController.php [hunk 9]
```diff
@@ -0,0 +1,63 @@
...
+  }
+
+  /**
+   * A simple method for displaying the Hello @Username.
+   *
+   * @return array
+   *   A render array to display the hello message with cache tags applied.
+   */
+  public function showName() {
+    $user = User::load(\Drupal::currentUser()->id());
```
Comment (vivekpan27) [inline web/modules/custom/mymodule/src/Controller/FirstController.php]: This is not the standard procedure of loading users.
Please check how we can load users using service.

File: web/modules/custom/mymodule/src/Form/FirstForm.php [hunk 10]
```diff
@@ -0,0 +1,106 @@
...
+use Drupal\Core\Form\FormBase;
+use Drupal\Core\Form\FormStateInterface;
+
+/**
+ * A basic form.
+ */
+class FirstForm extends FormBase {
+
+  /**
+   * {{@inheritdoc}}
```
Comment (vivekpan27) [inline web/modules/custom/mymodule/src/Form/FirstForm.php]: `{@inheritdoc}`

File: web/modules/custom/mymodule/src/Form/FirstForm.php (hunk 7 above)
Comment (vivekpan27) [reply web/modules/custom/mymodule/src/Form/FirstForm.php]: This is still not fixed.

File: web/modules/custom/mymodule/src/Form/FirstForm.php (hunk 6 above)
Comment (vivekpan27) [reply web/modules/custom/mymodule/src/Form/FirstForm.php:11]: Alright.

**Review by vivekpan27 — CHANGES_REQUESTED** [review]
Please fix.

**Comment by vivekpan27** [comment]
@Arijit-Acharya-innoraft Did you run the code through php code sniffer before submitting the PR?
Because the issues which remain would have been detected had you run it through PHPCS.

File: web/modules/custom/mymodule/src/Form/FirstForm.php (hunk 7 above)
Comment (Arijit-Acharya-innoraft) [reply web/modules/custom/mymodule/src/Form/FirstForm.php]: Updated.

File: web/modules/custom/mymodule/mymodule.info.yml (hunk 1 above)
Comment (Arijit-Acharya-innoraft) [reply web/modules/custom/mymodule/mymodule.info.yml]: Ok. Removing it.

File: web/modules/custom/mymodule/mymodule.module (hunk 8 above)
Comment (Arijit-Acharya-innoraft) [reply web/modules/custom/mymodule/mymodule.module]: By mistake it was there. Updated.

File: web/modules/custom/mymodule/src/Controller/FirstController.php (hunk 9 above)
Comment (Arijit-Acharya-innoraft) [reply web/modules/custom/mymodule/src/Controller/FirstController.php]: At that time, just started with custom module, wasn't that handy with dependency injection. Came to learn about it after the dependency injection ticket. Fixed it now.

File: web/modules/custom/mymodule/src/Form/FirstForm.php (hunk 10 above)
Comment (Arijit-Acharya-innoraft) [reply web/modules/custom/mymodule/src/Form/FirstForm.php]: fixed

**Comment by Arijit-Acharya-innoraft** [comment]
@vivekpan27  yes I do have run the phpcs Check Drupal coding standards, found some errors then ran the phpcs automatic fix code standard before submitting the PR. I thought the errors as "{{@iinheritdoc}}" would be rectified .The other suggested changes was not shown as error there.

**Comment by Arijit-Acharya-innoraft** [comment]
@vivekpan27  Fixed the suggested changes.

**Pull request merged**